api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -202,6 +206,26 @@ type HostedControlPlaneSpec struct {
 	Labels map[string]string `json:"labels,omitempty"`
 }
 
+// kubeAPICustomKubeconfig is a group that englobes an additional KAS server address and the customkubeconfig generated for the kube-apiserver.
+// +optional
+type KubeAPICustomKubeconfig struct {
+	// customKubeConfig specifies the name and key for the External Custom kubeconfig secret.
+	// When set, it triggers the generation of a secret with the specified name containing a kubeconfig within the `HostedCluster` namespace.
+	// This kubeconfig will also be referenced in the `HostedCluster.status` as `customkubeconfig` (remarking that in the HC status is lowercase meanwhile this one is camelCase to keep the consistency with the rest of the spec).
+	// If removed during day-2 operations, all related secrets and status references will also be deleted.
+	// +optional
```

Comment:
please add docs and cel validation that only pass if these fields are both set, unless other cases are supported.

Reply:
I don't think the validation is necessary at all:
- The user defines the `KasDNSName` in the HostedCluster
- The Controller grab that name and put it in the HostedControlPlane object inside of the `CustomKubeconfig` struct
- The Controller will also generate the Certificate as an Opaque `SecretRef`.

So this will be managed by us and not by the user and does not needs to be controlled by CEL.

Reply:
ok, so If this is only an implementation detail with well known values then, there's no need for it in the API at all.

Reply:
I'm followed the same approach than we do with `kubeconfig`, I was thinking this is necessary for ROSA or MCE in any way.

Reply:
To clarify, I've removed the hcp.spec.KasCustomKubeconfig because it's an old implementation detail that is not relevant anymore and I've changed the PR to simplify the implementation in that way.